packages/yoastseo/spec/parse/build/private/getTextElementPositionsSpec.js

```diff
@@ -371,13 +380,117 @@ describe( "A test for getting positions of sentences", () => {
 		const sentencesWithPositions = [ { text: "שלום עולם.", sourceCodeRange: { startOffset: 8, endOffset: 31 } },
 			{ text: "זה החתול שלי.", sourceCodeRange: { startOffset: 31, endOffset: 44 } } ];
 
-		expect( getSentencePositions( node, sentences ) ).toEqual( sentencesWithPositions );
+		expect( getTextElementPositions( node, sentences ) ).toEqual( sentencesWithPositions );
+	} );
+
+	it( "gets the token positions from a node that doesn't have descendants other than the Text node", function() {
+		// HTML: <p>Hello, world!</p>.
+		const node = new Paragraph( {}, [ { name: "#text", value: "Hello, world!" } ],
+			{
+				startOffset: 0,
+				endOffset: 21,
+				startTag: {
+					startOffset: 0,
+					endOffset: 3,
+				},
+				endTag: {
+					startOffset: 17,
+					endOffset: 21,
+				},
+			} );
+
+		const tokens = [ { text: "Hello" }, { text: "," }, { text: " " }, { text: "world" }, { text: "!" } ];
+		const tokensWithPositions = [
+			{ text: "Hello", sourceCodeRange: { startOffset: 3, endOffset: 8 } },
+			{ text: ",", sourceCodeRange: { startOffset: 8, endOffset: 9 } },
+			{ text: " ", sourceCodeRange: { startOffset: 9, endOffset: 10 } },
+			{ text: "world", sourceCodeRange: { startOffset: 10, endOffset: 15 } },
+			{ text: "!", sourceCodeRange: { startOffset: 15, endOffset: 16 } },
+		];
+
+		expect( getTextElementPositions( node, tokens ) ).toEqual( tokensWithPositions );
+	} );
+
+	it( "gets the token positions from a node that has multiple descendants", function() {
+		// HTML: <p><strong>Hello</strong>, <em>world</em>!</p>.
+
+		const node = new Paragraph( {}, [
+			{ name: "#text", value: "Hello, world!" },
+			{
+				name: "strong",
+				attributes: {},
+				childNodes: [
+					{
+						name: "#text",
+						value: "Hello",
+					},
+				],
+				sourceCodeLocation: {
+					startTag: {
+						startOffset: 3,
+						endOffset: 11,
+					},
+					endTag: {
+						startOffset: 16,
+						endOffset: 25,
+					},
+					startOffset: 3,
+					endOffset: 25,
+				},
+			},
+			{
+				name: "em",
+				attributes: {},
+				childNodes: [
+					{
+						name: "#text",
+						value: "world",
+					},
+				],
+				sourceCodeLocation: {
+					startTag: {
+						startOffset: 27,
+						endOffset: 31,
+					},
+					endTag: {
+						startOffset: 36,
+						endOffset: 41,
+					},
+					startOffset: 27,
+					endOffset: 41,
+				},
+			},
+		],
+		{
+			startTag: {
+				startOffset: 0,
+				endOffset: 3,
+			},
+			endTag: {
+				startOffset: 42,
+				endOffset: 46,
+			},
+			startOffset: 0,
+			endOffset: 46,
+		} );
+
+		const tokens = [ { text: "Hello" }, { text: "," }, { text: " " }, { text: "world" }, { text: "!" } ];
+		const tokensWithPositions = [
+			{ text: "Hello", sourceCodeRange: { startOffset: 3, endOffset: 25 } },
```

Comment:
This is in line with how it works for Sentences (see some of the other tests in this file), but I can imagine that we would want to have the `startOffset` be **after** the opening `<strong>` tag, and end before the closing `</strong>` tag.